python/ray/serve/tests/test_gcs_failure.py

```diff
@@ -127,6 +132,7 @@ def router_populated_with_replicas(
     else:
         replicas = get_replicas_func()
 
+    print(f"Replica set in router: {replicas}")
```

Comment:
Is replicas the only thing that would help to debug? We might want to consider adding logs for other thing like `threshold`, `handle`, and if handle exist, maybe router as well

Reply:
Currently the issue is the wait for condition is passing yet for some reason the behavior is as if the replicas haven't been broadcasted correctly - so I think replicas is what I'm most curious about, will add other things if necessary!